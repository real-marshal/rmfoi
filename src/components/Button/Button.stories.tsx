import type { Meta, Story } from '@storybook/react'
import { Button, ButtonProps } from './Button'

export default {
  title: 'Common/Button',
  component: Button,
} as Meta

const Template: Story<ButtonProps> = (args) => <Button {...args} />

export const Basic = Template.bind({})
Basic.args = {
  children: 'Button',
}

export const Secondary = Template.bind({})
Secondary.args = {
  children: 'Button',
  secondary: true,
}
