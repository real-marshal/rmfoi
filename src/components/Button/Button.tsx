import type { MouseEventHandler, ReactNode } from 'react'
import styled from '@emotion/styled'
import { dt } from '@/features/Theming'

interface ButtonProps {
  children: ReactNode
  onClick?: MouseEventHandler<HTMLButtonElement>
}

const StyledButton = styled.button`
  background: ${dt.colors.gradient};
  padding: ${dt.padding.lg} ${dt.padding.xl};
  border: none;
  border-radius: 10px;
  cursor: pointer;
`

const Button = ({ children, onClick }: ButtonProps) => (
  <StyledButton onClick={onClick}>{children}</StyledButton>
)

export { Button }
export type { ButtonProps }
