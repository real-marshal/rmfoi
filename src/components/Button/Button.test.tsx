import { composeStories } from '@storybook/react'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { Button } from './Button'
import * as stories from './Button.stories'

const testStories = Object.values(composeStories(stories)).map(
  (Story) => [Story.storyName, Story] as const
)

describe('Button component', () => {
  it.each(testStories)('renders %s story', (_, Story) => {
    const view = render(<Story />)
    expect(view.baseElement).toMatchSnapshot()
  })

  it('calls onClick with event when clicked', async () => {
    const user = userEvent.setup()
    const clickHandler = jest.fn()

    render(<Button onClick={clickHandler}>Click me daddy</Button>)

    await user.click(screen.getByRole('button', { name: 'Click me daddy' }))

    expect(clickHandler).toHaveBeenCalled()
  })
})
