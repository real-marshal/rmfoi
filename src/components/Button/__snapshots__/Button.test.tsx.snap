// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders Basic story 1`] = `
.emotion-0 {
  background: var(--dt-colors-gradient);
  padding: var(--dt-padding-lg) var(--dt-padding-xl);
  border: none;
  border-radius: 10px;
  cursor: pointer;
}

<body>
  <div>
    <button
      class="emotion-0 emotion-1"
    >
      Button
    </button>
  </div>
</body>
`;
